### DISCUSSION TITLE
How use predict_one if X is 3d-data?

### DISCUSSION THREAD

**Discussion by xayam** [online-ml/river#1201] [Q&A] [answered]
```python
import numpy as np
from river import naive_bayes
from river import stream


X = np.array([[[-1, -1], [-1, -1]], [[1, -1], [-1, -1]], [[-3, -1], [-1, -1]], [[-1, -1], [-1, -1]],
              [[-1, -1], [-1, -1]], [[-1, 1], [-2, -1]]])
Y = np.array([1, 1, 1, 2, 2, 2])

model = naive_bayes.GaussianNB()

for x, y in stream.iter_array(X, Y):
    model.learn_one(x, y)

print(model.predict_one(???))
```

How use model.predict_one if X is 3d-list?

**Comment by MaxHalford** [comment, marked as answer]
Hey there. We don't support this paradigm out of the box. You would have to convert your raw inputs into features. The features would depend on the nature of your raw inputs, and what you're trying to model.

**Reply by xayam** [reply]
can you give an example of such a function if i want to process a checkerboard view? what does it look like syntactically?

**Reply by MaxHalford** [reply]
In short, your goal is simply to build a function that converts a checkerboard representation (your 3D structure) into a dictionary. Dictionaries are the inputs River users as features. I suggest you familiarize yourself with River.

Sorry but I don't have time to help you more than that. I have no context on what you're trying to do, and that is out of scope of this library. The problem you're facing is not a problem with this library.

**Reply by xayam** [reply]
thanks

**Reply by MaxHalford** [reply]
No problem. Good luck with your problem! Please mark this thread as answered if you think we're done here :)
